woocommerce/woocommerce
# Add order attribution install banner in core analytics overview

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Part of #51567, this PR adds the order attribution install banner in the core analytics overview page. The banner will be shown if:
- The `woocommerce-analytics` plugin is not installed yet, and
- The merchant hasn't clicked the `Dismiss` button.

https://github.com/user-attachments/assets/ace17a90-007d-45cc-811d-b712ebd3a300

Note that when clicking `Try it now` on the banner, it will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found as the plugin hasn't been published.

#### Event tracking

This PR also adds three events for tracking:
- `order_attribution_install_banner_viewed`
- `order_attribution_install_banner_clicked`
- `order_attribution_install_banner_dismissed`

#### Next step

- Implement displaying the banner for % of merchants
- Display the banner in other places within WooCommerce Admin
- The changelog entry will be added in the final PR once all the banners are added



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Do not install `woocommerce-analytics` plugin
2. Go to `Analytics -> Overview`
3. It should show the banner
4. Click `Try it now`
5. It will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found
6. Click `Dismiss`
7. The banner is gone
8. Check the option `woocommerce_order_attribution_install_banner_dismissed` in `wp_options` table
9. The value should be `yes`
10. Delete the option `woocommerce_order_attribution_install_banner_dismissed`
11. Go to `Analytics -> Overview`, the banner should be shown again
12. Install `woocommerce-analytics` plugin, do not activate it
13. Go to `Analytics -> Overview`, the banner should not be shown
14. Activate `woocommerce-analytics` plugin
15. Go to `Analytics -> Overview`, the banner should not be shown 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
